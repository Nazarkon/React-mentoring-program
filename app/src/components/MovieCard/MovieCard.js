import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './MovieCard.scss';
import { createMovieItemsList } from '../../helpers/MovieItemsHelpers';

const MovieCard = ({ id, imageUrl, name, year, genreList, handleClick }) => {
  const [imageLoadError, setImageLoadError] = useState(false);

  const handleImageError = () => {
    setImageLoadError(true);
  };

  return (
    <div className="movie-card-container" onClick={() => handleClick(id)}>
      <div className="movie-card-container-image">
        {imageLoadError ? (
          <div className="movie-card-error" alt="error" src={'../../assets/images/404-error.png'} />
        ) : (
          <img
            className="movie-card-image"
            alt="film poster"
            src={imageUrl}
            onError={handleImageError}
          />
        )}
      </div>
      <div className="movie-card-container-title">
        <h4 className="movie-card-title">{name}</h4>
        <p className="movie-card-year">{year}</p>
      </div>
      <span className="movie-card-genres">{createMovieItemsList(genreList)}</span>
    </div>
  );
};

MovieCard.propTypes = {
  id: PropTypes.number.isRequired,
  imageUrl: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  year: PropTypes.string.isRequired,
  genreList: PropTypes.array.isRequired,
  handleClick: PropTypes.func
};

export default MovieCard;
