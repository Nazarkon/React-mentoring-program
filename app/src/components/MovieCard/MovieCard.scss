.movie-card-container{
    width: 100%;
    cursor: pointer;
    max-width: 332px;
    display: flex;
    justify-content: center;
    flex-direction: column;
      .movie-card-container-image{
        .movie-card-image{
            width: 100%;
            height: 100%;
            min-height: 500px;
        }
        .movie-card-error{
          width: 100%;
          height: 100%;
          display: flex;
          justify-content: center;
          min-height: 500px;
          background-image: url('../../assets/images/404-error.png');
          background-size: contain;
          background-repeat: no-repeat;
        }
      }
      .movie-card-container-title{
        display: flex;
        justify-content: space-between;
        align-items: center;
            .movie-card-title{
              color: #FFFFFF;
              font-size: 18px;
              flex: 1;
            }
            .movie-card-year{
                border: 1px solid #FFFFFF;
                border-radius: 4px;
                padding: 4px 8px;
                font-size: 18px;
                color: #FFFFFF;
            }
      }
        .movie-card-genres{
          color: #FFFFFF;
        }
}