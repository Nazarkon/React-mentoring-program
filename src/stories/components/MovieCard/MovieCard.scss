.movie-card-container{
    width: 100%;
    max-width: 332px;
    display: flex;
    justify-content: center;
    background: #424242;
    flex-direction: column;
      .movie-card-container-image{
        .movie-card-image{
            width: 100%;
            height: 100%;
        }
      }
      .movie-card-container-title{
        display: flex;
        justify-content: space-between;
        align-items: center;
            .movie-card-title{
              color: #FFFFFF;
              font-size: 18px;
            }
            .movie-card-year{
                border: 1px solid #FFFFFF;
                border-radius: 4px;
                padding: 4px 8px;
                font-size: 18px;
                color: #FFFFFF;
            }
      }
        .movie-card-genres{
          color: #FFFFFF;
        }
}