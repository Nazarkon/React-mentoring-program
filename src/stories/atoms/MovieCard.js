import React from "react";
import "../assets/styles/MovieCard.scss";

const MovieCard = ({ imageUrl, name, year, genreList }) => {

    const createListOfGenres = () => {
        if(!genreList || !genreList.length) return '-';

        return genreList.join('&');
    }
    return (
        <div className="movie-card-container">
            <div className="movie-card-container-image">
                <img className="movie-card-image" alt="film poster" src={imageUrl}/>
            </div>
            <div className="movie-card-container-title">
                <h4 className="movie-card-title">{name}</h4>
                <p className="movie-card-year">{year}</p>
            </div>
            <span className="movie-card-genres">{createListOfGenres()}</span>
        </div>
    )
}

export default MovieCard