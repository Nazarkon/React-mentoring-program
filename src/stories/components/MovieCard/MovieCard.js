import React from "react";
import "./MovieCard.scss";
import { createMovieItemsList } from '../../../helpers/MovieItemsHelpers'

const MovieCard = ({ imageUrl, name, year, genreList, handleClick }) => {

    return (
        <div className="movie-card-container" onClick={(e) => handleClick(e)}>
            <div className="movie-card-container-image">
                <img className="movie-card-image" alt="film poster" src={imageUrl}/>
            </div>
            <div className="movie-card-container-title">
                <h4 className="movie-card-title">{name}</h4>
                <p className="movie-card-year">{year}</p>
            </div>
            <span className="movie-card-genres">{createMovieItemsList(genreList)}</span>
        </div>
    )
}

export default MovieCard