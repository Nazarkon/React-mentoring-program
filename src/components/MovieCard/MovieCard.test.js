/* eslint-disable testing-library/prefer-screen-queries */
import { render } from '@testing-library/react';
import '@testing-library/jest-dom'
import MovieCard from "./MovieCard";


describe("MovieCard component", () => {

    const mockData = {
        image: 'Bitmap.png',
        name: 'Pulp Fiction',
        year: '2003',
        genreList: ["Action", "Adventure"]
    }

    it("matches snapshot", () => {
        const { container } = render(
        <MovieCard 
            imageUrl={mockData.image}
            name={mockData.name} 
            year={mockData.year} 
            genreList={mockData.genreList}
        />);
        expect(container).toMatchSnapshot();
      });
    it("Check that component renders movie information correctly", () => {
        const { getByAltText, getByText } = render(
            <MovieCard 
            imageUrl={mockData.image}
            name={mockData.name} 
            year={mockData.year} 
            genreList={mockData.genreList}
        />
        );

        expect(getByAltText("film poster")).toHaveAttribute(
            "src",
            "Bitmap.png"
          );
        
        expect(getByText("Pulp Fiction")).toBeInTheDocument();

        expect(getByText("2003")).toBeInTheDocument();

        expect(getByText("Action&Adventure")).toBeInTheDocument();
        
    });
});