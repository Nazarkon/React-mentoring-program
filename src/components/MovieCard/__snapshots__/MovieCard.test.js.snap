// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard component matches snapshot 1`] = `
<div>
  <div
    class="movie-card-container"
  >
    <div
      class="movie-card-container-image"
    >
      <div
        class="movie-card-container-menu"
      />
      <img
        alt="film poster"
        class="movie-card-image"
        src="Bitmap.png"
      />
    </div>
    <div
      class="movie-card-container-title"
    >
      <h4
        class="movie-card-title"
      >
        Pulp Fiction
      </h4>
      <p
        class="movie-card-year"
      >
        2003
      </p>
    </div>
    <span
      class="movie-card-genres"
    >
      Action&Adventure
    </span>
  </div>
</div>
`;
